transcom/mymove
# INT B-19902 move gets locked when office user enters and UI changes

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-19902)

## Summary

In order to "lock" moves while office users are working on them, we must initially set that lock and then display the relevant UI elements to inform the users that the move is currently being worked on at a glance (when viewing their queue) as well as who is working on the move (after entering the move).

This PR does the following:
- updating all database queries done when searching and viewing queues to pull in foreign key values from the `office_users` table 
- updating swagger definitions with newly introduced values
- created new service object to lock a move with office user that is locking it and 30 minute expiration timer
- created component for `LockedMoveBanner.jsx`
- added lock icon to all queues that conditionally renders on values in move
- wrapped all relevant UI code in feature flag checks for `move_code` feature flag

### How to test

1. In your `envrc`, turn `FEATURE_FLAG_MOVE_LOCK` to `true`
2. Fire up `client` & `server`
3. Log in as an office user (can use multi role for efficiency)
4. Enter any move as TOO/TIO/SC
5. Log out
6. Log back in as another user
7. When your queue loads, you should be able to see a lock icon to the left of the move that the previous user entered (we are simulating them "working on it")
8. Enter the move
9. You should see a banner displayed at the top of the screen of the first & last name of the user that is "working on it" as well as their transpo office
10. The lock icon & banner will display until the time in `lock_expires_at` passes, but if any office user enters the move after that time, it will be reset to 30 minutes. Future work will be done to clear these values when the user returns to their queue. Right now they will stay in the database, but UI elements will conditionally render.
11. Within each office role, view queues, perform searches, and the lock icons should still be visible in the queue - if the lock icon is on the move, the banner should display if you are NOT the user that has it locked

## Screenshots
TOO view with locked moves
![Screenshot 2024-05-09 at 2 51 20 PM](https://github.com/transcom/mymove/assets/136510600/e48a76f8-0136-475f-8212-59b2fda3f176)

TOO search with locked move
![Screenshot 2024-05-09 at 2 51 32 PM](https://github.com/transcom/mymove/assets/136510600/0e7a3ae0-a8e7-4128-9edd-e28fa598035b)

TIO with locked moves
![Screenshot 2024-05-09 at 2 51 43 PM](https://github.com/transcom/mymove/assets/136510600/4d2dfd1f-8146-4dd4-a878-f95c076798c0)

SC with locked moves in counseling queue & PPM closeout queue
![Screenshot 2024-05-09 at 2 50 59 PM](https://github.com/transcom/mymove/assets/136510600/1e87154a-de37-488a-bc95-aa5000650b82)

![Screenshot 2024-05-09 at 2 52 01 PM](https://github.com/transcom/mymove/assets/136510600/77ebc334-5577-4cfb-ab0e-2748d6d87f8e)

Banner that displays when user is viewing a locked move - read only permissions are being handled in later stories
![Screenshot 2024-05-09 at 2 52 35 PM](https://github.com/transcom/mymove/assets/136510600/8c8435f0-776b-4bb0-b352-312d235e6740)
